engine/src/main/java/org/terasology/particles/updating/ParticleUpdaterImpl.java

```diff
@@ -299,29 +302,39 @@ private void updateEmitter(final ParticleEmitterComponent particleEmitter, final
         }
     }
 
-    private void updateParticleSystem(final ParticleEmitterComponent partSys, final float delta) {
-        if (partSys.enabled && (partSys.particleSpawnsLeft == ParticleEmitterComponent.INFINITE_PARTICLE_SPAWNS || partSys.particleSpawnsLeft > 0)) {
-            updateEmitter(partSys, 0, delta); // Emit particles
+    private void updateParticleEmitters(final ParticleEmitterComponent emitter, final float delta) {
+        if (emitter.enabled && (emitter.particleSpawnsLeft == ParticleEmitterComponent.INFINITE_PARTICLE_SPAWNS || emitter.particleSpawnsLeft > 0)) {
+            updateEmitter(emitter, 0, delta); // Emit particles
         }
 
-        updateParticles(partSys, delta); // Update particle lifetime and Affectors
+        updateEmitterLifeTime(emitter, delta);
+    }
+
+    private void updateParticleData(final ParticleEmitterComponent particleSystem, float delta) {
```

Comment:
Same here. In combination with the comment further up this might also be the right spot to describe invariants or assumptions.